android/java/org/chromium/chrome/browser/crypto_wallet/fragments/PortfolioFragment.java

```diff
@@ -56,6 +76,7 @@ private EthJsonRpcController getEthJsonRpcController() {
     public void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setHasOptionsMenu(true);
+        getEthJsonRpcController().addObserver(this);
```

Comment:
Why do we need that observer? Only we control when we change network. And we send that event in `public void onItemSelected(AdapterView<?> parent, View view, int position, long id) {`. We can just hook update balance there